### PR TITLE
Add a client class

### PR DESCRIPTION
**Pull request by djmetzle**
Add a client class.

### CODE REVIEW

File: lib/client/client.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      OpenStruct.new(:code => response.code, :body => body)
+   rescue JSON::ParserError
+      response
+   end
+
+   def request(method, path, params)
+      case method
+      when :get
+         full_path = encode_path_params(path, params)
+         request = VERB_MAP[method.to_sym].new(full_path)
```
Comment (MooreGuy) [inline lib/client/client.rb]: string to lower so that "GET" or "get" will work?

File: lib/client/client.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+   def request_json(method, path, params)
+      response = request(method, path, params)
+      body = JSON.parse(response.body)
+
+      OpenStruct.new(:code => response.code, :body => body)
+   rescue JSON::ParserError
+      response
+   end
+
+   def request(method, path, params)
```
Comment (g1y) [inline lib/client/client.rb:33]: method to lower so it "GET" or "get" would work?

**Comment by g1y** [comment]
CR :+1:

**Comment by djmetzle** [comment]
Merging.

**Pull request merged**